docs/theory/diffusion/diffusion.rst

```diff
@@ -0,0 +1,715 @@
+
+    def simulate_with_exponential_length(tau, params, num_tracks=10, min_length=5):
+        """Simulate tracks of varying lengths
+
+        The lengths are drawn from an exponential distribution."""
+        # Redraw tracks until we get 20 above the minimum length
+        lengths = np.zeros(shape=num_tracks, dtype=int)
+        elements_below = lengths < min_length
+        while np.any(elements_below):
+            elements_below = lengths < min_length
+            lengths[elements_below] = np.round(np.random.exponential(tau, size=np.sum(elements_below)) / params["dt"]).astype(int)
+
+            group = lk.simulation.simulate_diffusive_tracks(**params, steps=max(5, lengths[0]), num_tracks=1)
+            for num_points in lengths[1:]:
+                group += lk.simulation.simulate_diffusive_tracks(**params, steps=max(5, num_points), num_tracks=1)
+
+        return group
+
+    def simulate_tracks(snr):
+        """Function used to simulate tracks"""
+        params = snr_to_diffusion_parameters(snr)
+        tracks = [
+            simulate_with_exponential_length(1.1, params, num_tracks=20)
+            for _ in range(100)
+        ]
+        return tracks, params["diffusion_constant"]
+
+    def simple_average_est(tracks, method):
+        """Just "average" the results for the tracks (NOT a good method)"""
+        diffusion_estimates = tracks.estimate_diffusion(method)
+        avg_diff_est = np.mean([d.value for d in diffusion_estimates])
+        std_diff_est = np.std([d.value for d in diffusion_estimates]) / np.sqrt(len(diffusion_estimates))
+        return CustomDiffusionEstimate(avg_diff_est, std_diff_est)
+
+    methods = {
+        "cve single": lambda list_of_tracks: [t.estimate_diffusion("cve") for tracks in list_of_tracks for t in tracks],
+        "gls single": lambda list_of_tracks: [t.estimate_diffusion("gls") for tracks in list_of_tracks for t in tracks],
+        "ols single": lambda list_of_tracks: [t.estimate_diffusion("ols") for tracks in list_of_tracks for t in tracks],
+        "cve unweighted": lambda list_of_tracks: [simple_average_est(t, "cve") for t in list_of_tracks],
+        "cve ensemble": lambda list_of_tracks: [t.ensemble_diffusion("cve") for t in list_of_tracks],
+        "ols ensemble": lambda list_of_tracks: [t.ensemble_diffusion("ols") for t in list_of_tracks],
+    }
+
+    snrs = 10**np.arange(-1, 1.1, 0.125)
+    results, variances = test_estimation_methods(simulate_tracks, snrs, methods=methods)
+
+    plt.figure(figsize=(12, 4))
+    plt.subplot(1, 2, 1)
+    for method in ("ols single", "gls single", "cve single"):
+        plot_accuracy(snrs, results[method], variance=variances[method], label=method);
+    plt.ylim([-1, 3])
+
+    plt.subplot(1, 2, 2)
+    for method in ("cve unweighted", "cve ensemble", "ols ensemble"):
+        plot_accuracy(snrs, results[method], variance=variances[method], label=method);
+    plt.ylim([-1, 3])
```

Comment:
might be nice to put some titles here to make it more clear what the two plots are comparing

Reply:
👍